lib/node/runner/api_node.go

```diff
@@ -124,10 +125,10 @@ func (api NetworkHandlerNode) MessageHandler(w http.ResponseWriter, r *http.Requ
 	}
 
 	if err = common.RunChecker(checker, common.DefaultDeferFunc); err != nil {
-		if _, ok := err.(common.CheckerErrorStop); ok {
-			return
+		if len(checker.Transaction.H.Hash) > 0 {
+			block.SaveTransactionHistory(api.storage, checker.Transaction, body, block.BlockTransactionHistoryStatusRejected)
 		}
-		httputils.WriteJSONError(w, err)
+		http.Error(w, err.Error(), httputils.StatusCode(err))
```

Comment:
It response just like other node APIs
The response may be handled and wrapped with JSON if needed.

Reply:
You mean that `err.Error()` just returns JSON string or plaintext string?
And response's header such as `Content-Type`? :->